pkg/controller/jobframework/reconciler.go

```diff
@@ -368,7 +369,29 @@ func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Objec
 	if owner.Name != object.GetName() {
 		return false
 	}
-	return job.EquivalentToWorkload(*wl)
+
+	jobPodSets := job.PodSets()
+
+	if !workload.CanBePartiallyAdmitted(wl) || job.IsSuspended() {
+		// the two sets should fully match.
+		return utilequality.ComparePodSetSlices(jobPodSets, wl.Spec.PodSets, true)
+	}
+
+	// If the job is not suspended, the podSet counts of the job could be lower that
+	// the initial values, in case of partial admission.
+	if !utilequality.ComparePodSetSlices(jobPodSets, wl.Spec.PodSets, false) {
+		return false
+	}
+
+	// Make sure the job podSets don't have higher counts, and use more resources that it originally
+	// requested.
+	// the ComparePodSetSlices fails if the lengths are not equal, don't check it once more
+	for i := range jobPodSets {
+		if jobPodSets[i].Count > wl.Spec.PodSets[i].Count || jobPodSets[i].Count < pointer.Int32Deref(wl.Spec.PodSets[i].MinCount, wl.Spec.PodSets[i].Count) {
```

Comment:
can we check that the count is equal to what's in `wl.Status.Admission`?

Reply:
The count in admission can be smaller in case of partial reclaim.